File: frontend/src/main/web/src/app/services/board.service.ts
import { Injectable } from '@angular/core';
import { Board } from '../models/Board';
import { ApiService } from './api.service';

@Injectable({
  providedIn: 'root'
})
export class BoardService {

  constructor(private apiService: ApiService) { }

  public async createBoard(title: string) {
    const body = {
      title,
    };
    return this.apiService.post<Board>('/board', body);
  }

  public async getBoardById(id: string): Promise<Board> {
    return this.apiService.get<Board>('/board', { id })
  }

  public async getAllBoards(): Promise<Board[]> {
    return this.apiService.get<Board[]>('/board/all');
  }

  public async updateBoard(id: string, title: string): Promise<Board> {
    const body = {
      id,
      title
    };
    return this.apiService.post('/board/update', body)
  }
}

File: backend/src/main/java/com/manuelsch/simplekanban/service/BoardService.java
package com.manuelsch.simplekanban.service;

import com.manuelsch.simplekanban.DTOs.exceptionHandling.RecordNotFoundException;
import com.manuelsch.simplekanban.models.Board;

import java.util.List;

/**
 * Business logic for dealing with Kanban boards
 */
public interface BoardService {

    /**
     * Creates a new Kanban board
     *
     * @param title
     * @return
     * @throws AssertionError
     */
    Board createBoard(String title);

    /**
     * Returns the Kanban board with the given ID
     *
     * @param id
     * @return
     */
    Board getBoardById(String id) throws RecordNotFoundException;

    /**
     * Returns all Kanban boards stored in the database
     *
     * @return
     */
    List<Board> getAllBoards();

}
